realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -78,9 +85,14 @@
     protected Row row;
     protected Realm realm;
 
+    private final List<RealmChangeListener> listeners = new CopyOnWriteArrayList<RealmChangeListener>();
```

Comment:
I talked to @emanuelez and @kneth in the office and I summarised our thinking in #1036